### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'filename'

### ISSUE THREAD

**Issue by edmoartist** [hako-mikan/sd-webui-supermerger#259]
I always get this error when trying to merge LORAs

Traceback (most recent call last):
  File "C:\Users\61429\STABES\sd.webui\webui\extensions\sd-webui-supermerger\scripts\mergers\pluslora.py", line 257, in lmerge
    ln.append(c_lora.filename)
AttributeError: 'NoneType' object has no attribute 'filename'

Let me know how to fix this
Cheers,

**Comment by hako-mikan** [comment]
The error is due to the LoRA you specified not being registered in the web-ui. However, since I have designed it to display and select from a list of registered LoRAs, such a problem shouldn't occur. Are you using any special characters or strings?

**Comment by edmoartist** [comment]
> The error is due to the LoRA you specified not being registered in the web-ui. However, since I have designed it to display and select from a list of registered LoRAs, such a problem shouldn't occur. Are you using any special characters or strings?

no special characters or strings as far as I know, I did make these LoRAs myself though with Kohya but I cant see that effecting it. thanks for getting back to me

**Comment by hako-mikan** [comment]
Does this issue also occur with LoRA downloaded from civitai or elsewhere?
If it does happen, please let me know the specific LoRA in question.

**Comment by ExissNA** [comment]
> Does this issue also occur with LoRA downloaded from civitai or elsewhere? ...

I also encountered this issue, and posted thinking it was an automatic1111 issue.  All of my LoRa's were downloaded direct from civitai.  It seemed to be random when it would have issues.  Sometimes using the same LoRa would work, or make the error.  When this happens, I am unable to generate images after that, and have to shut down and reboot the web-ui server.

**Comment by ExissNA** [comment]
> Is this error still occurring now?

Didn't mean to abandon this but I have not tried again recently.  I did update, but just have not tried it.  I will do so hopefully this weekend and report back.

**Comment by ExissNA** [comment]
I managed to run 3 merges back to back without issue.  I have not tried LoRas.

**Comment by ema7569** [comment]
I think it's possible that because when clicking "from prompt" it use the lora name show in prompt and not the lora name on disk. They can be different. I also remember I already have see in lora properties a lora name that can be different from the lora filename and supermerger seem to require the lora filename to work.

**Comment by 57331Fury** [comment]
I was getting the same issue, but it seems that for me at least, ema7569`s answer was correct. Since I have many loras for many different base models, I have taken to adding "Pony", "ILL", SDXL" etc if they aren`t already in the filename, so getting the disk name, rather than the prompt name worked.

**Comment by BlackSheep1983** [comment]
Yes this is caused by loras having different output and filename, not a real issue I think..
